### ISSUE TITLE
Problem of different predicted results

### ISSUE THREAD

**Issue by kasyful** [xinwucwp/faultSeg#20]
Dear Dr @llmpass and Dr @xinwucwp,

I would say this work is great contribution to fault interpretation study.

I try to recreate and build a similar model (same training dataset and same architecture), however when i inference the model to my datasets, it seems to detect bright amplitude, rather than fault itself.

![image](https://github.com/xinwucwp/faultSeg/assets/91114474/d890f790-fb2f-491e-90c0-ba4210509ee0)

I am genuinely appreciate if you could provide any ideas or suggestion why the model work this way.

Thanks.

**Comment by llmpass** [comment]
Could you normalize your image, say minus mean and divide by standard deviation before you performing the inference?

**Comment by kasyful** [comment]
Thank you for your prompt response @llmpass ,

I did tried both z-score and minmax scaler normalization, it showing almost similar results.

![image](https://github.com/xinwucwp/faultSeg/assets/91114474/c254e746-81c4-408e-b09b-5bc8d5ce4835)

My apologies, DataGenerator function already performed z-score for the training datasets. I still couldnt understand why the model predict differently.

Ive go thru each Unet architecture and cross entrophy balanced but still couldnt figure it out.

Thank you for your help.

**Comment by llmpass** [comment]
Yes, that's strange. I'm sorry that I cannot help you too much this time.

**Comment by suporange** [comment]
> Thank you for your prompt response @llmpass ,
> 
> I did tried both z-score and minmax scaler normalization, it showing almost similar results.
> 
> ![image](https://private-user-images.githubusercontent.com/91114474/311614272-c254e746-81c4-408e-b09b-5bc8d5ce4835.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1OTcyODEsIm5iZiI6MTcxMjU5Njk4MSwicGF0aCI6Ii85MTExNDQ3NC8zMTE2MTQyNzItYzI1NGU3NDYtODFjNC00MDhlLWIwOWItNWJjOGQ1Y2U0ODM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA4VDE3MjMwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTczMzgxYjNkOTk5ZWYyZDk1NzQzOTljZjdiNWY2YTNlMzg3ZmRmZWZlMDNlYWIwY2ExOTlhODU4NmI2YTVlYjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bzfXbvusvyd1QrxXvDmtXAuJhEDwk_6B7Zy-KqwOex4)
> 
> My apologies, DataGenerator function already performed z-score for the training datasets. I still couldnt understand why the model predict differently.
> 
> Ive go thru each Unet architecture and cross entrophy balanced but still couldnt figure it out.
> 
> Thank you for your help.

Sir, I met the same problem. Do you have any ideas?

**Comment by suporange** [comment]
> Thank you for your prompt response @llmpass ,
> 
> I did tried both z-score and minmax scaler normalization, it showing almost similar results.
> 
> ![image](https://private-user-images.githubusercontent.com/91114474/311614272-c254e746-81c4-408e-b09b-5bc8d5ce4835.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1OTcyODEsIm5iZiI6MTcxMjU5Njk4MSwicGF0aCI6Ii85MTExNDQ3NC8zMTE2MTQyNzItYzI1NGU3NDYtODFjNC00MDhlLWIwOWItNWJjOGQ1Y2U0ODM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA4VDE3MjMwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTczMzgxYjNkOTk5ZWYyZDk1NzQzOTljZjdiNWY2YTNlMzg3ZmRmZWZlMDNlYWIwY2ExOTlhODU4NmI2YTVlYjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bzfXbvusvyd1QrxXvDmtXAuJhEDwk_6B7Zy-KqwOex4)
> 
> My apologies, DataGenerator function already performed z-score for the training datasets. I still couldnt understand why the model predict differently.
> 
> Ive go thru each Unet architecture and cross entrophy balanced but still couldnt figure it out.
> 
> Thank you for your help.

@kasyful  Sir, i have met the same problem. Do you have any ideas?

**Comment by kasyful** [comment]
> > Thank you for your prompt response @llmpass ,
> > I did tried both z-score and minmax scaler normalization, it showing almost similar results.
> > ![image](https://private-user-images.githubusercontent.com/91114474/311614272-c254e746-81c4-408e-b09b-5bc8d5ce4835.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1OTcyODEsIm5iZiI6MTcxMjU5Njk4MSwicGF0aCI6Ii85MTExNDQ3NC8zMTE2MTQyNzItYzI1NGU3NDYtODFjNC00MDhlLWIwOWItNWJjOGQ1Y2U0ODM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA4VDE3MjMwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTczMzgxYjNkOTk5ZWYyZDk1NzQzOTljZjdiNWY2YTNlMzg3ZmRmZWZlMDNlYWIwY2ExOTlhODU4NmI2YTVlYjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bzfXbvusvyd1QrxXvDmtXAuJhEDwk_6B7Zy-KqwOex4)
> > My apologies, DataGenerator function already performed z-score for the training datasets. I still couldnt understand why the model predict differently.
> > Ive go thru each Unet architecture and cross entrophy balanced but still couldnt figure it out.
> > Thank you for your help.
> 
> @kasyful Sir, i have met the same problem. Do you have any ideas?

@suporange , you need to transpose the dataset before inference the model.
